testsuite/AdResS/FreeEnergyCompensation/test_FreeEnergyCompensation.py

```diff
@@ -0,0 +1,133 @@
+import espressopp
+import mpi4py.MPI as MPI
+
+import unittest
+
+class TestFreeEnergyCompensation(unittest.TestCase):
+    def setUp(self):
+        # set up system
+        system = espressopp.System()
+        box = (10, 10, 10)
+        system.bc = espressopp.bc.OrthorhombicBC(system.rng, box)
+        system.skin = 0.3
+        system.comm = MPI.COMM_WORLD
+        nodeGrid = espressopp.tools.decomp.nodeGrid(espressopp.MPI.COMM_WORLD.size)
+        cellGrid = espressopp.tools.decomp.cellGrid(box, nodeGrid, 1.5, 0.3)
+        system.storage = espressopp.storage.DomainDecompositionAdress(system, nodeGrid, cellGrid)
+        self.system = system
+
+    def test_slab(self):
+        # add some particles
+        particle_list = [
+            (1, 1, 0, espressopp.Real3D(5.5, 5.0, 5.0), 1.0, 0),
+            (2, 1, 0, espressopp.Real3D(6.5, 5.0, 5.0), 1.0, 0),
+            (3, 1, 0, espressopp.Real3D(7.5, 5.0, 5.0), 1.0, 0),
+            (4, 1, 0, espressopp.Real3D(8.5, 5.0, 5.0), 1.0, 0),
+            (5, 1, 0, espressopp.Real3D(9.5, 5.0, 5.0), 1.0, 0),
+            (6, 0, 0, espressopp.Real3D(5.5, 5.0, 5.0), 1.0, 1),
+            (7, 0, 0, espressopp.Real3D(6.5, 5.0, 5.0), 1.0, 1),
+            (8, 0, 0, espressopp.Real3D(7.5, 5.0, 5.0), 1.0, 1),
+            (9, 0, 0, espressopp.Real3D(8.5, 5.0, 5.0), 1.0, 1),
+            (10, 0, 0, espressopp.Real3D(9.5, 5.0, 5.0), 1.0, 1),
+        ]
+        tuples = [(1,6),(2,7),(3,8),(4,9),(5,10)]
+        self.system.storage.addParticles(particle_list, 'id', 'type', 'q', 'pos', 'mass','adrat')
+        ftpl = espressopp.FixedTupleListAdress(self.system.storage)
+        ftpl.addTuples(tuples)
+        self.system.storage.setFixedTuplesAdress(ftpl)
+        self.system.storage.decompose()
+
+        # generate a verlet list
+        vl = espressopp.VerletListAdress(self.system, cutoff=1.5, adrcut=1.5,
+                                dEx=2.0, dHy=1.0, adrCenter=[5.0, 5.0, 5.0], sphereAdr=False)
+
+        # initialize lambda values
+        integrator = espressopp.integrator.VelocityVerlet(self.system)
+        integrator.dt = 0.01
+        adress = espressopp.integrator.Adress(self.system,vl,ftpl)
+        integrator.addExtension(adress)
+        espressopp.tools.AdressDecomp(self.system, integrator)
+
+        # set up FEC
+        fec = espressopp.integrator.FreeEnergyCompensation(self.system, center=[5.0, 5.0, 5.0])
+        fec.addForce(itype=3,filename="table_fec.tab",type=1)
+        integrator.addExtension(fec)
+
+        # x coordinates of particles before integration
+        before = [self.system.storage.getParticle(i).pos[0] for i in range(1,6)]
+
+        # run ten steps and compute energy
+        integrator.run(10)
+        energy = fec.computeCompEnergy()
+
+        # x coordinates of particles after integration
+        after = [self.system.storage.getParticle(i).pos[0] for i in range(1,6)]
+
+        # run checks
+        self.assertAlmostEqual(before[0], after[0], places=5)
```

Comment:
I don't understand what's happening here. Why are some of the particles moving and others not?